// viem/_esm/_test/generated.js
// Generated by @wagmi/cli@0.1.6 on 11/09/2023 at 9:45:29 am
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// IERC1271Wallet
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export const ierc1271WalletABI = [
    {
        stateMutability: 'view',
        type: 'function',
        inputs: [
            { name: 'hash', internalType: 'bytes32', type: 'bytes32' },
            { name: 'signature', internalType: 'bytes', type: 'bytes' },
        ],
        name: 'isValidSignature',
        outputs: [{ name: 'magicValue', internalType: 'bytes4', type: 'bytes4' }],
    },
];
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// VerifySig
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export const verifySigABI = [
    {
        stateMutability: 'nonpayable',
        type: 'constructor',
        inputs: [
            { name: '_signer', internalType: 'address', type: 'address' },
            { name: '_hash', internalType: 'bytes32', type: 'bytes32' },
            { name: '_signature', internalType: 'bytes', type: 'bytes' },
        ],
    },
    {
        stateMutability: 'nonpayable',
        type: 'function',
        inputs: [
            { name: '_signer', internalType: 'address', type: 'address' },
            { name: '_hash', internalType: 'bytes32', type: 'bytes32' },
            { name: '_signature', internalType: 'bytes', type: 'bytes' },
        ],
        name: 'isValidUniversalSig',
        outputs: [{ name: '', internalType: 'bool', type: 'bool' }],
    },
];
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// ERC20InvalidTransferEvent
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export const erc20InvalidTransferEventABI = [
    {
        type: 'event',
        anonymous: false,
        inputs: [
            { name: 'from', internalType: 'address', type: 'address', indexed: true },
            { name: 'to', internalType: 'address', type: 'address', indexed: false },
            {
                name: 'value',
                internalType: 'uint256',
                type: 'uint256',
                indexed: false,
            },
        ],
        name: 'Transfer',
    },
    {
        stateMutability: 'nonpayable',
        type: 'function',
        inputs: [
            { name: 'recipient', internalType: 'address', type: 'address' },
            { name: 'amount', internalType: 'uint256', type: 'uint256' },
        ],
        name: 'transfer',
        outputs: [],
    },
];
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// GH434
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export const gh434ABI = [
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'bar',
        outputs: [{ name: '', internalType: 'string', type: 'string' }],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'baz',
        outputs: [{ name: '', internalType: 'uint256', type: 'uint256' }],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'foo',
        outputs: [
            { name: 'a', internalType: 'uint256', type: 'uint256' },
            { name: 'b', internalType: 'bool', type: 'bool' },
        ],
    },
];
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// EnsAvatarTokenUri
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export const ensAvatarTokenUriABI = [
    {
        stateMutability: 'view',
        type: 'function',
        inputs: [{ name: 'tokenId', internalType: 'uint256', type: 'uint256' }],
        name: 'ownerOf',
        outputs: [{ name: '', internalType: 'address', type: 'address' }],
    },
    {
        stateMutability: 'view',
        type: 'function',
        inputs: [{ name: 'tokenId', internalType: 'uint256', type: 'uint256' }],
        name: 'tokenURI',
        outputs: [{ name: '', internalType: 'string', type: 'string' }],
    },
];
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// ErrorsExample
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export const errorsExampleABI = [
    {
        type: 'error',
        inputs: [
            {
                name: 'foo',
                internalType: 'struct ErrorsExample.Foo',
                type: 'tuple',
                components: [
                    { name: 'sender', internalType: 'address', type: 'address' },
                    { name: 'bar', internalType: 'uint256', type: 'uint256' },
                ],
            },
            { name: 'message', internalType: 'string', type: 'string' },
            { name: 'number', internalType: 'uint256', type: 'uint256' },
        ],
        name: 'ComplexError',
    },
    {
        type: 'error',
        inputs: [{ name: 'message', internalType: 'string', type: 'string' }],
        name: 'SimpleError',
    },
    { type: 'error', inputs: [], name: 'SimpleErrorNoArgs' },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'assertRead',
        outputs: [],
    },
    {
        stateMutability: 'nonpayable',
        type: 'function',
        inputs: [],
        name: 'assertWrite',
        outputs: [],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'complexCustomRead',
        outputs: [],
    },
    {
        stateMutability: 'nonpayable',
        type: 'function',
        inputs: [],
        name: 'complexCustomWrite',
        outputs: [],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'divideByZeroRead',
        outputs: [{ name: '', internalType: 'uint256', type: 'uint256' }],
    },
    {
        stateMutability: 'nonpayable',
        type: 'function',
        inputs: [],
        name: 'divideByZeroWrite',
        outputs: [{ name: '', internalType: 'uint256', type: 'uint256' }],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'overflowRead',
        outputs: [{ name: '', internalType: 'uint256', type: 'uint256' }],
    },
    {
        stateMutability: 'nonpayable',
        type: 'function',
        inputs: [],
        name: 'overflowWrite',
        outputs: [{ name: '', internalType: 'uint256', type: 'uint256' }],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'requireRead',
        outputs: [],
    },
    {
        stateMutability: 'nonpayable',
        type: 'function',
        inputs: [],
        name: 'requireWrite',
        outputs: [],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'revertRead',
        outputs: [],
    },
    {
        stateMutability: 'nonpayable',
        type: 'function',
        inputs: [],
        name: 'revertWrite',
        outputs: [],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'simpleCustomRead',
        outputs: [],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [],
        name: 'simpleCustomReadNoArgs',
        outputs: [],
    },
    {
        stateMutability: 'nonpayable',
        type: 'function',
        inputs: [],
        name: 'simpleCustomWrite',
        outputs: [],
    },
];
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// OffchainLookupExample
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export const offchainLookupExampleABI = [
    {
        stateMutability: 'nonpayable',
        type: 'constructor',
        inputs: [{ name: 'urls_', internalType: 'string[]', type: 'string[]' }],
    },
    {
        type: 'error',
        inputs: [
            { name: 'sender', internalType: 'address', type: 'address' },
            { name: 'urls', internalType: 'string[]', type: 'string[]' },
            { name: 'callData', internalType: 'bytes', type: 'bytes' },
            { name: 'callbackFunction', internalType: 'bytes4', type: 'bytes4' },
            { name: 'extraData', internalType: 'bytes', type: 'bytes' },
        ],
        name: 'OffchainLookup',
    },
    {
        stateMutability: 'view',
        type: 'function',
        inputs: [{ name: 'name', internalType: 'string', type: 'string' }],
        name: 'getAddress',
        outputs: [{ name: '', internalType: 'address', type: 'address' }],
    },
    {
        stateMutability: 'view',
        type: 'function',
        inputs: [
            { name: 'result', internalType: 'bytes', type: 'bytes' },
            { name: 'extraData', internalType: 'bytes', type: 'bytes' },
        ],
        name: 'getAddressWithProof',
        outputs: [{ name: '', internalType: 'address', type: 'address' }],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [
            {
                name: '_ethSignedMessageHash',
                internalType: 'bytes32',
                type: 'bytes32',
            },
            { name: '_signature', internalType: 'bytes', type: 'bytes' },
        ],
        name: 'recoverSigner',
        outputs: [{ name: '', internalType: 'address', type: 'address' }],
    },
    {
        stateMutability: 'pure',
        type: 'function',
        inputs: [{ name: 'sig', internalType: 'bytes', type: 'bytes' }],
        name: 'splitSignature',
        outputs: [
            { name: 'r', internalType: 'bytes32', type: 'bytes32' },
            { name: 's', internalType: 'bytes32', type: 'bytes32' },
            { name: 'v', internalType: 'uint8', type: 'uint8' },
        ],
    },
    {
        stateMutability: 'view',
        type: 'function',
        inputs: [{ name: '', internalType: 'uint256', type: 'uint256' }],
        name: 'urls',
        outputs: [{ name: '', internalType: 'string', type: 'string' }],
    },
];
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
// Payable
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
export const payableABI = [
    {
        stateMutability: 'payable',
        type: 'function',
        inputs: [],
        name: 'pay',
        outputs: [],
    },
];
